### PR TITLE
8255352: Archive important test outputs in submit workflow

### PR DESCRIPTION
**Pull request by shipilev**
Currently, we are only archiving `build/*/test-results`. But hs_errs, replays, and test outputs are actually in `build/*/test-support`! Which means once any test fails, we only have the summary of the run, not the bits that would actually help to diagnose the problem. 

Archiving the entire test-support seems too much, it yields 50K artifacts available as 1 GB zip archive in GH GUI. On top of that `upload-artifact` does the upload per-file, which takes tens of minutes on 50K files (and I suspect many API calls).

But we can pick up important test outputs selectively and then also compress them. See sample artifact here:
 https://github.com/shipilev/jdk/runs/1301540541

It packages the final ZIP like this:

```
$ unzip -t test-results_.zip
 Archive:  test-results_.zip
    testing: artifact/                OK
    testing: artifact/linux-x64-debug_testresults_hs_tier1_common.zip   OK
    testing: artifact/linux-x64-debug_testsupport_hs_tier1_common.zip   OK
    ...
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x32 | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |  ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8255352](https://bugs.openjdk.java.net/browse/JDK-8255352): Archive important test outputs in submit workflow

### Reviewers
 * [Robin Westberg](https://openjdk.java.net/census#rwestberg) (@rwestberg - Committer) ⚠️ Review applies to 7584cbcd045403f994b5a0b72bda87b524cfa235
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/842/head:pull/842`
`$ git checkout pull/842`

### CODE REVIEW

File: .github/workflows/submit.yml [hunk 1]
```diff
@@ -734,12 +761,41 @@ jobs:
         if: always()
         run: echo ("logsuffix=" + ("${{ matrix.test }}" -replace "/", "_" -replace " ", "_")) | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8
 
-      - name: Persist test logs
+      - name: Package test results
+        if: always()
+        working-directory: build/run-test-prebuilt/test-results/
+        run: >
+          $env:Path = "$HOME\cygwin\cygwin64\bin;$HOME\cygwin\cygwin64\bin;$env:Path" ;
```
Comment (rwestberg) [inline .github/workflows/submit.yml:768]:
```suggestion
          $env:Path = "$HOME\cygwin\cygwin64\bin;$env:Path" ;
```

File: .github/workflows/submit.yml [hunk 2]
```diff
@@ -734,12 +761,41 @@ jobs:
...
+          zip -r9
+          "$HOME/windows-x64${{ matrix.artifact }}_testresults_${{ env.logsuffix }}.zip"
+          .
+        continue-on-error: true
+
+      - name: Package test support
+        if: always()
+        working-directory: build/run-test-prebuilt/test-support/
+        run: >
+          $env:Path = "$HOME\cygwin\cygwin64\bin;$HOME\cygwin\cygwin64\bin;$env:Path" ;
```
Comment (rwestberg) [inline .github/workflows/submit.yml:778]:
```suggestion
          $env:Path = "$HOME\cygwin\cygwin64\bin;$env:Path" ;
```

**Review by rwestberg — APPROVED** [review]
Looks good! I also tried including the entire test-support folder at some point, but as you noticed it takes quite a bit of time and space. So this approach seems much better.

Minor comment: The cygwin\bin folder is included twice in PATH on Windows to work around a bug in JTReg (the regex used to detect cygwin does not match the very first entry in the PATH list. Perhaps I should file a bug for that..) but it's not really needed if you are executing something else.

**Comment by shipilev** [comment]
Thanks! I have not even noticed there is double item in `PATH`.

**Review by magicus — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**